store/action-queue.html

```diff
@@ -8,22 +8,21 @@
 	window.D2L.Siren.ActionQueue = {
 		queueEnd: Promise.resolve(),
 
-		// Task needs to be a function that returns a promise
 		enqueue: function(task) {
-			var taskRunner;
-			var promise = new Promise(function(resolve, reject) {
-				taskRunner = function() {
-					try {
-						var result = task();
-						result.then(resolve).catch(reject);
-					} catch (e) {
-						reject(e);
-					}
-				};
+			var queuedResolve;
+			var promise = new Promise(function(resolve) {
+				queuedResolve = resolve;
 			});
-			this.queueEnd.then(taskRunner).catch(taskRunner);
+
+			this.queueEnd.then(runTask, runTask);
 			this.queueEnd = promise;
 			return promise;
+
+			function runTask() {
+				queuedResolve(new Promise(function(resolve) {
+					resolve(task());
```

Comment:
Probably showing my lack of chops when it comes to more complex promise handling here,  but do these changes mean that the original promise never rejects? What happens in the call to `resolve` if `task()` rejects?

Reply:
If a `Promise` initializer throws, it will reject. This is approximately the same as doing `try { resolve(task()); catch (e) { resolve(Promise.reject(e)); } ` or similar.

`resolve(42)` means its a promise to the value 42
`resolve(anotherPromise)` means it takes on whatever that other promise it, whether it fulfills or rejects